lib/go/Makefile

```diff
@@ -51,6 +52,8 @@ PROTOGEN_ARCH := amd64
 ifeq (i386,$(PROTOC_ARCH))
 PROTOC_ARCH := x86_32
 PROTOGEN_ARCH := 386
+else ifeq (arm64, $(PROTOC_ARCH))
+PROTOC_ARCH := x86_64
```

Comment:
Let's fix this properly. Hacking arm64 -> x86_64 just breaks arm builds on Linux. protoc 3.20.1 and later support arm64 on Mac, and should be backwards compatible.

Some other changes we need WRT versioning:

PROTOC_OS and PROTOC_ARCH need some massaging, as the releases don't match exactly what `uname -a` / `uname -s` return. Can we do something lik:

```
-# Fix OS string for Mac builds.
+# Fix OS string for Mac and Linux builds.
 PROTOC_OS := $(shell uname -s)
 PROTOGEN_OS := $(shell uname -s)
 ifeq (Darwin,$(PROTOC_OS))
 PROTOC_OS := osx
 endif
+ifeq (Linux,$(PROTOC_OS))
+PROTOC_OS := linux
+endif
 
-# Allow building on 32 bit machines.
+# Fix architecture naming.
 PROTOC_ARCH := $(shell uname -m)
 PROTOGEN_ARCH := amd64
 ifeq (i386,$(PROTOC_ARCH))
 PROTOC_ARCH := x86_32
 PROTOGEN_ARCH := 386
 else ifeq (arm64, $(PROTOC_ARCH))
-PROTOC_ARCH := x86_64
+PROTOC_ARCH := aarch_64
+else ifeq (aarch64, $(PROTOC_ARCH))
+PROTOC_ARCH := aarch_64
 endif
```

Reply:
protoc 3.20.1 produces identical Go output to 3.16.0 so there's no issue in using the newer version. I've verified this locally on linux/aarch_64. This doesn't change any runtime or compile-time dependencies for our code. Let's update the version to 3.20.1.